src/flows/context/flow.query.tsx

```diff
@@ -278,6 +280,58 @@ export const FlowQueryProvider: FC = ({children}) => {
     status = RemoteDataState.Loading
   }
 
+  const queryDependents = (startID: string) => {
+    sendEvent(notebookQueryKey)
+    _queryDependents(startID)
+  }
+
+  const _queryDependents = (startID: string) => {
+    if (status === RemoteDataState.Loading) {
+      return
+    }
+
+    let map = generateMap(runMode === RunMode.Run)
+
+    if (!map.length) {
+      return
+    }
+
+    map = map.slice(map.indexOf(map.find(m => m.id === startID)))
```

Comment:
this feels way more complicated than it needs to be. Maybe i'm just confused, why not just get an id reference and slice from there rather than do the find, index, slice? Something like:

```
const index = map.indexOf(m => m.id === startID)
if (index === -1) {
  ... not sure what we'd wanna do here
} else {
  map = map.slice(index)
}

Reply:
ooh an opportunity for me to use findIndex :)